docs/dev/windows.md

```diff
@@ -622,133 +690,138 @@ input
+contains the one to be applied after the window function. Elements evicted before the application of the window 
+function will not be processed by it.
+
+Flink comes with three pre-implemented evictors. These are:
+
+* `CountEvictor`: keeps up to a user-specified number of elements from the window and discards the remaining ones from 
+the beginning of the window buffer.
+* `DeltaEvictor`: takes a `DeltaFunction` and a `threshold`, computes the delta between the last element in the 
+window buffer and each of the remaining ones, and removes the ones with a delta greater or equal to the threshold.
+* `TimeEvictor`: takes as argument an `interval` in milliseconds and for a given window, it finds the maximum 
+timestamp `max_ts` among its elements and removes all the elements with timestamps smaller than `max_ts - interval`.
 
-The internal `Trigger` API is still considered experimental but you can check out the code
-if you want to write your own custom trigger:
-{% gh_link /flink-streaming-java/src/main/java/org/apache/flink/streaming/api/windowing/triggers/Trigger.java "Trigger.java" %}.
+<span class="label label-info">Default</span> By default, all the pre-implemented evictors apply their logic before the 
+window function.
 
-## Non-keyed Windowing
+<span class="label label-danger">Attention</span> Specifying an evictor prevents any pre-aggregation, as all the 
+elements of a window have to be passed to the evictor before applying the computation.
 
-You can also leave out the `keyBy()` when specifying a windowed transformation. This means, however,
-that Flink cannot process windows for different keys in parallel, essentially turning the
-transformation into a non-parallel operation.
+<span class="label label-danger">Attention</span> Flink provides no guarantees about the order of the elements within
+a window. This implies that although an evictor may remove elements from the beginning of the window, these are not 
+necessarily the ones that arrive first or last.
 
-<span class="label label-danger">Warning</span> As mentioned in the introduction, non-keyed
-windows have the disadvantage that work cannot be distributed in the cluster because
-windows cannot be computed independently per key. This can have severe performance implications.
 
+## Allowed Lateness
 
-The basic structure of a non-keyed windowed transformation is as follows:
+When working with *event-time* windowing it can happen that elements arrive late, *i.e.* the watermark that Flink uses to 
+keep track of the progress of event-time is already past the end timestamp of a window to which an element belongs. See 
+[event time](./event_time.html) and especially [late elements](./event_time.html#late-elements) for a more thorough 
+discussion of how Flink deals with event time.
+
+By default, late elements are dropped if their associated window was already evaluated. However, 
```

Comment:
They are not dropped when the window was already evaluated but when the watermark is past the end of the window plus the allowed lateness.